include/circt/Dialect/RTL/Statements.td

```diff
@@ -51,3 +51,30 @@ def ReadInOutOp : RTLOp<"read_inout", [NoSideEffect]> {
 
   let verifier = [{ return ::verifyReadInOutOp(*this); }];
 }
+
+
+def StructExplodeOp : RTLOp<"struct_explode", [NoSideEffect]> {
```

Comment:
While this is nicely definable, I'd recommend dropping this until/if we have a specific use-case for it.  This will be a pain to support in the verilog printer, and it is just sugar for a bunch of extracts.  Let's stick with a simpler representation and add sugar later if needed.

Reply:
FWIW, exploding to variables is actually simple in verilog, consider:
```
module main;

typedef struct packed {
    logic [7:0] a;
    logic [31:0] b;
} s1;

  initial
    begin
      s1 foo = {8'd1,32'd2};
      logic[7:0] c;
      logic[31:0] d;
      {c,d} = foo;
      $display(foo.a);
       $display(foo.b);
      $display(c);
      $display(d);
      $finish ;
    end
endmodule
```

Reply:
Oh I didn't know that was legal, I retract my concern then! Thanks!